github/github.go

```diff
@@ -735,11 +735,14 @@ func category(path string) rateLimitCategory {
 // Deprecated: RateLimit is deprecated, use RateLimits instead.
 func (c *Client) RateLimit() (*Rate, *Response, error) {
 	limits, resp, err := c.RateLimits()
+	if err != nil {
+		return nil, resp, err
+	}
 	if limits == nil {
-		return nil, nil, err
+		return nil, resp, fmt.Errorf("RateLimits returned nil limits and error; unable to extract Core rate limit")
 	}
 
```

Comment:
Nitpick, but I would remove this blank line now. It does not provide value at this point IMO.

Without the blank line separating them, it would be a little more visible that `limits` is checked for being non-`nil` before dereferencing `limits.Core`.